projects/scion/microfrontend-platform/src/lib/client/messaging/messaging.spec.ts

```diff
@@ -1300,33 +1377,66 @@ describe('Messaging', () => {
     it('should complete the source observable when all subscribers unsubscribed', async () => {
       await MicrofrontendPlatform.startHost({applications: []});
 
-      const subscription = Beans.get(MessageClient).observe$('some-topic').subscribe();
-      await waitUntilSubscriberCount('some-topic', 1);
+      const subscription1 = Beans.get(MessageClient).observe$('some-topic').subscribe();
+      const subscription2 = Beans.get(MessageClient).observe$('some-topic').subscribe();
+      await waitUntilSubscriberCount('some-topic', 2);
 
-      const testee = new Subject<void>()
+      const captor = new ObserveCaptor();
+      new Subject<void>()
         .pipe(
           takeUntilUnsubscribe('some-topic'),
           timeoutWith(new Date(Date.now() + 2000), throwError('[SpecTimeoutError] Timeout elapsed.')),
         )
-        .toPromise();
+        .subscribe(captor);
+
+      // unsubscribe subscription1
+      subscription1.unsubscribe();
+      await waitUntilSubscriberCount('some-topic', 1);
+
+      expect(captor.hasCompleted()).withContext('hasCompleted').toBeFalse();
+      expect(captor.getValues()).withContext('emissions').toEqual([]);
 
-      // unsubscribe from the topic
-      subscription.unsubscribe();
+      // unsubscribe subscription2
+      subscription2.unsubscribe();
+      await waitUntilSubscriberCount('some-topic', 0);
 
-      await expectPromise(testee).toResolve();
+      expect(captor.hasCompleted()).withContext('hasCompleted').toBeTrue();
+      expect(captor.getValues()).withContext('emissions').toEqual([]);
     });
 
     it('should complete the source observable immediately when no subscriber is subscribed', async () => {
       await MicrofrontendPlatform.startHost({applications: []});
 
-      const testee = new Subject<void>()
+      const captor = new ObserveCaptor();
+      new Subject<void>()
         .pipe(
           takeUntilUnsubscribe('nobody-subscribed-to-this-topic'),
           timeoutWith(new Date(Date.now() + 500), throwError('[SpecTimeoutError] Timeout elapsed.')),
         )
-        .toPromise();
+        .subscribe(captor);
 
-      await expectPromise(testee).toResolve();
+      await waitUntilSubscriberCount('some-topic', 0);
+      expect(captor.hasCompleted()).withContext('hasCompleted').toBeTrue();
+      expect(captor.getValues()).withContext('emissions').toEqual([]);
+    });
+
+    it('should not complete the source Observable upon plaform shutdown (as per API)', async () => {
+      await MicrofrontendPlatform.startHost({applications: []});
+
+      // GIVEN
+      const captor = new ObserveCaptor();
+      new Subject<void>()
+        .pipe(
+          takeUntilUnsubscribe('nobody-subscribed-to-this-topic'),
+          timeoutWith(new Date(Date.now() + 500), throwError('[SpecTimeoutError] Timeout elapsed.')),
```

Comment:
If the source observable is not completed, why the timeout error is not thrown?

Reply:
The error is actually thrown but has no effect on the expectation. The use of the `timeout` operator in this and the above test makes no sense, so I removed it. Instead, I cancel the subscription after the expectation.